python/ray/log.py

```diff
@@ -0,0 +1,158 @@
+import logging
+import re
+import sys
+from logging.config import dictConfig
+
+
+class ContextFilter(logging.Filter):
+    """A filter that adds ray context info to log records.
+
+    This filter adds a package name to append to the message as well as information
+    about what worker emitted the message, if applicable.
+    """
+
+    logger_regex = re.compile(r"ray(\.(?P<subpackage>\w+))?(\..*)?")
+    package_message_names = {
+        "air": "AIR",
+        "data": "Data",
+        "rllib": "RLlib",
+        "serve": "Serve",
+        "train": "Train",
+        "tune": "Tune",
+        "workflow": "Workflow",
+    }
+
+    def __init__(self, *args, **kwargs):
+        super().__init__(*args, **kwargs)
+        self.global_worker_mode = None
+
+    def filter(self, record: logging.LogRecord) -> bool:
+        """Add context information to the log record.
+
+        This filter adds a package name from where the message was generated as
+        well as the worker IP address, if applicable.
+
+        Args:
+            record: Record to be filtered
+
+        Returns:
+            True if the record is to be logged, False otherwise. (This filter only
+            adds context, so records are always logged.)
+        """
+        match = self.logger_regex.search(record.name)
+        if match and match["subpackage"] in self.package_message_names:
+            record.package = f"[Ray {self.package_message_names[match['subpackage']]}]"
+        else:
+            record.package = ""
+
+        # Lazily load ray when this is called to avoid circular import
+        import ray
+
+        if hasattr(ray, "_private"):
+            if self.global_worker_mode is None:
+                self.global_worker_mode = ray._private.worker.global_worker.mode
+
+            if (
+                self.global_worker_mode
+                and self.global_worker_mode == ray._private.worker.WORKER_MODE
+            ):
+                record.node_ip_address = (
+                    ray.runtime_context.get_runtime_context().worker.node_ip_address
```

Comment:
remove it? Seems like it is unused?

Reply:
Yes, this is left over from splitting this PR into two. Thanks for catching this!

Reply:
Hmm this still seems to be not removed yet? Is it expected?